### DISCUSSION TITLE
dnsServer.Prepare: dns: proxy.ParseUpstreamsConfig: wrong hostname specification   ？？？

### DISCUSSION THREAD

**Discussion by Potterli20** [AdguardTeam/AdGuardHome#2819] [Q&A] [answered]
错误日记AdGuardHome.err：
2021/03/15 01:53:46 [info] Service control action: run
2021/03/15 01:53:46 [info] AdGuard Home, version v0.105.2, channel release, arch linux amd64
2021/03/15 01:53:46 [info] AdGuard Home is running as a service
2021/03/15 01:53:46 [info] Initializing auth module: /opt/AdGuardHome/data/sessions.db
2021/03/15 01:53:46 [info] Auth: initialized.  users:1  sessions:2
2021/03/15 01:53:46 [info] Initialize web module
2021/03/15 01:53:46 [fatal] dnsServer.Prepare: dns: proxy.ParseUpstreamsConfig: wrong hostname specification: ##

配置文件AdGuardHome.yaml：
bind_host: 0.0.0.0
bind_port: 81
beta_bind_port: 3001
users:
- name: potter
  password: 
http_proxy: ""
language: zh-cn
rlimit_nofile: 0
debug_pprof: false
web_session_ttl: 720
dns:
  bind_host: 0.0.0.0
  port: 54
  statistics_interval: 90
  querylog_enabled: true
  querylog_file_enabled: true
  querylog_interval: 90
  querylog_size_memory: 1000
  anonymize_client_ip: true
  protection_enabled: true
  blocking_mode: default
  blocking_ipv4: ""
  blocking_ipv6: ""
  blocked_response_ttl: 10
  parental_block_host: family-block.dns.adguard.com
  safebrowsing_block_host: standard-block.dns.adguard.com
  ratelimit: 0
  ratelimit_whitelist: []
  refuse_any: true
  upstream_dns:
  - https://dns.alidns.com/dns-query
  upstream_dns_file: /root/domain_full.txt
  bootstrap_dns:
  - 9.9.9.10
  - 149.112.112.10
  - 2620:fe::10
  - 2620:fe::fe:10
  all_servers: true
  fastest_addr: false
  allowed_clients: []
  disallowed_clients: []
  blocked_hosts:
  - version.bind
  - id.server
  - hostname.bind
  cache_size: 0
  cache_ttl_min: 0
  cache_ttl_max: 0
  bogus_nxdomain: []
  aaaa_disabled: false
  enable_dnssec: true
  edns_client_subnet: true
  max_goroutines: 0
  ipset: []
  filtering_enabled: true
  filters_update_interval: 1
  parental_enabled: false
  safesearch_enabled: false
  safebrowsing_enabled: false
  safebrowsing_cache_size: 0
  safesearch_cache_size: 0
  parental_cache_size: 0
  cache_time: 0
  rewrites:
  - domain: pixiv.net
    answer: www.pixiv.net
  - domain: twitter.com
    answer: www.twitter.com
  - domain: google.com
    answer: www.google.com
  - domain: error.baidu.com
    answer: baidu.com
  - domain: twitch.tv
    answer: www.twitch.tv
  - domain: '*.douban.com'
    answer: forward.douban.com
  - domain: youtube.com
    answer: www.youtube.com
  - domain: data.bilibili.com
    answer: interface.biliapi.com
  - domain: fonts.googleapis.com
    answer: fonts.gstatic.com
  - domain: fonts.gstatic.com
    answer: fonts.loli.net
  - domain: ajax.googleapis.com
    answer: ajax.loli.net
  - domain: themes.googleusercontent.com
    answer: themes.loli.net
  - domain: cdnjs.cloudflare.com
    answer: cdnjs.loli.net
  - domain: secure.gravatar.com
    answer: gravatar.loli.net
  - domain: translate.googleapis.com
    answer: translation.googleapis.com
  - domain: baidu.com
    answer: www.baidu.com
  blocked_services: []
  customresolver: null
tls:
  enabled: false
  server_name: ""
  force_https: false
  port_https: 443
  port_dns_over_tls: 853
  port_dns_over_quic: 784
  port_dnscrypt: 0
  dnscrypt_config_file: ""
  allow_unencrypted_doh: false
  strict_sni_check: false
  certificate_chain: ""
  private_key: ""
  certificate_path: ""
  private_key_path: ""
filters:
- enabled: false
  url: http://dns.trli.club/dns/ad-hosts.txt
  name: ad-hosts
  id: 1612381151
- enabled: true
  url: http://dns.trli.club/dns/hosts.txt
  name: hosts
  id: 1612381152
- enabled: false
  url: http://dns.trli.club/dns/ad-adblock.txt
  name: ad-adblock.txt
  id: 1612381153
whitelist_filters:
- enabled: false
  url: https://raw.githubusercontent.com/Potterli20/filtering/master/filtering.txt
  name: potter
  id: 1612439425
user_rules:
- '@@||jump.alibaba.com^'
dhcp:
  enabled: false
  interface_name: ""
  dhcpv4:
    gateway_ip: ""
    subnet_mask: ""
    range_start: ""
    range_end: ""
    lease_duration: 86400
    icmp_timeout_msec: 1000
    options: []
  dhcpv6:
    range_start: ""
    lease_duration: 86400
    ra_slaac_only: false
    ra_allow_slaac: false
clients: []
log_compress: false
log_localtime: false
log_max_backups: 0
log_max_size: 500
log_max_age: 3
log_file: ""
verbose: false
schema_version: 7

文件加速domain_full.txt：
dns.trli.club/dns/dns-txt/gfwlist2adguardhome/blacklist_domain_full.txt

**Comment by ameshkov** [comment, marked as answer]
Happens due to invalid syntax, there're sometimes `##` instead of a hostname:
![image](https://user-images.githubusercontent.com/5947035/111084537-f07cea80-8523-11eb-88f3-b2a14cfe8102.png)

Also, are you absolutely sure that this list should be THAT huge?

There're many redundant domains like `dnsotls-ds.metric.gstatic.com` -- they all can be changed to `gstatic.com`.
And so on, it seems that the list can be compressed 10 times.

**Comment by Potterli20** [comment]
OK, I see where the mistake is. Thank you
